### PR TITLE
storage: fix that some latches of expired commands are not cleaned

### PR DESCRIPTION
**Pull request by sticnarf**
### What problem does this PR solve?

When a request expires at acquiring latches, it will yield an error immediately and release its latches. But previously, in the `release` method of `Latch`, it requires an accurate `owned_count`. Then, some locks inserted into the lock are not cleared and then blocks subsequent requests.

### What is changed and how it works?

If the deadline is exceeded, we first update the `owned_count` of the command. Then, calling `release` method can release all locks of the previous commands.

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/storage/txn/scheduler.rs [hunk 1]
```diff
@@ -250,7 +250,14 @@ impl<L: LockManager> SchedulerInner<L> {
         let tctx = task_slot.get_mut(&cid).unwrap();
         // Check deadline early during acquiring latches to avoid expired requests blocking
         // other requests.
-        tctx.task.as_ref().unwrap().deadline.check()?;
+        if let Err(e) = tctx.task.as_ref().unwrap().deadline.check() {
```
Comment (youjiali1995) [inline src/storage/txn/scheduler.rs:253]: The function name is misleading...

**Review by youjiali1995 — APPROVED** [review]

**Comment by youjiali1995** [comment]
Build failure...

**Review by ekexium — APPROVED** [review]

**Pull request merged**